packages/taquito/src/operations/operation-emitter.ts

```diff
@@ -251,7 +254,7 @@ export abstract class OperationEmitter {
     const errors = flattenErrors(results);
 
     if (errors.length) {
-      throw new TezosOperationError(errors);
+      throw new TezosOperationError(errors, 'Error occurred during operation preparation');
```

Comment:
We can specify that the error happens during the simulation of the validation of the operation